core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -280,9 +282,36 @@ protected ClassLoader createClassLoader(List<File> paths, ClassLoader parent, At
     }
 
     /**
+     * Get the list of all plugins that have ever been detached from Jenkins core.
+     * @return A {@link List} of {@link DetachedPlugin}s.
+     * @since FIXME
+     */
+    public static @Nonnull List<DetachedPlugin> getDetachedPlugins() {
+        return DETACHED_LIST;
+    }
+
+    /**
+     * Get the list of plugins that have been detached since a specific Jenkins release version.
+     * @param since The Jenkins version.
+     * @return A {@link List} of {@link DetachedPlugin}s.
+     * @since FIXME
+     */
+    public static @Nonnull List<DetachedPlugin> getDetachedPlugins(@Nonnull VersionNumber since) {
+        List<DetachedPlugin> detachedPlugins = new ArrayList<>();
+
+        for (DetachedPlugin detachedPlugin : DETACHED_LIST) {
+            if (!detachedPlugin.getSplitWhen().isOlderThan(since)) {
+                detachedPlugins.add(detachedPlugin);
+            }
+        }
+
+        return detachedPlugins;
+    }
+
+    /**
      * Information about plugins that were originally in the core.
      */
-    private static final class DetachedPlugin {
+    public static final class DetachedPlugin {
```

Comment:
Not if it is going to be returned from `getDetachedPlugins`, in which case you need a `@since` on this too.